fuel/src/test/kotlin/com/github/kittinunf/fuel/InterceptorTest.kt

```diff
@@ -304,15 +305,311 @@ class InterceptorTest : BaseTestCase() {
+            { r: Request ->
+                requests.add(r)
+                next(r)
+            }
+        }
+
+        val (originalRequest, _, result) = manager.request(Method.POST,
+                "http://httpstat.us/308")
+                .responseString()
+
+        val (data, error) = result
+
+        assertThat(data, notNullValue())
+        assertThat(error, nullValue())
+        assertThat(originalRequest.method, isEqualTo(Method.POST))
+        assertThat(requests[1].method, isEqualTo(Method.POST))
+    }
+
+    @Test
+    fun testHeaderIsPassingAlongWithRedirection() {
+        val manager = FuelManager()
+
+        val (_, _, result) = manager.request(Method.GET,
+                "http://httpbin.org/redirect-to",
+                listOf("url" to "http://httpbin.org/get"))
+                .header("Foo" to "bar")
+                .responseString()
+
+        val (data, error) = result
+
+        assertThat(data, notNullValue())
+        assertThat(data, containsString("http://httpbin.org/get"))
+        assertThat(data, containsString("\"Foo\":\"bar"))
+        assertThat(error, nullValue())
+    }
+
+    @Test
+    fun testHeaderIsPassingAlongWithRedirectionOnlySubPath() {
+        val manager = FuelManager()
+
+        val (_, _, result) = manager.request(Method.GET,
+                "http://httpbin.org/redirect-to",
+                listOf("url" to "/basic-auth/user/pass"))
+                .header("Foo" to "bar")
+                .responseString()
+
+        val (data, error) = result
+
+        assertThat(data, notNullValue())
+        assertThat(data, containsString("http://httpbin.org/get"))
+        assertThat(data, containsString("\"Foo\":\"bar"))
+        assertThat(error, nullValue())
+    }
+
+    @Test
+    fun testHeaderAuthenticationWillBeRemoveIfRedirectToDifferentHost() {
+        val manager = FuelManager()
+
+        val requests = mutableListOf<Request>()
+
+        manager.addRequestInterceptor { next: (Request) -> Request ->
+            { r: Request ->
+                requests.add(r)
+                next(r)
+            }
+        }
+
+        val (_, _, result) = manager.request(Method.GET,
+                "http://httpbin.org/redirect-to",
+                listOf("url" to "http://httpstat.us"))
+                .authenticate("foo", "bar")
+                .header("Foo" to "bar")
+                .responseString()
+
+
+        val (data, error) = result
+
+        assertThat(data, notNullValue())
+        assertThat(error, nullValue())
+        assertThat(requests[1].headers["Foo"], notNullValue())
+        assertThat(requests[1].headers["Authorization"], nullValue())
+    }
+
+    @Test
+    fun testDoNotAllowRedirect() {
+        val manager = FuelManager()
+
+        val (_, _, result) = manager.request(Method.GET,
+                "http://httpbin.org/redirect-to",
+                listOf("url" to "/get"))
+                .allowRedirect(false)
```

Comment:
@SleeplessByte this is a flag to stop follow the redirection. Do you think this makes sense?

Reply:
Yah, anything like that works fine. Since it will follow up to max number of redirects, you might want to change it to plural. Can you set that at manager level as well?

Reply:
Will work on the plural! 

> Can you set that at manager level as well?

I have thought about it and decided (by myself) that it is more flexible to do at the request level. Do you see any potential misuse or inconvenience?

Reply:
People are lazy; but it's soooo trivial to pipe requests (using extension methods) through my own set to always set allowRedirect(false) or whatever, so I really couldn't care less. I think it's a fine design decision; as long as it's documented how to disallow redirects :)